src/language_server_protocol.rs

```diff
@@ -1504,13 +1504,43 @@ impl LanguageClient {
         if help.signatures.is_empty() {
             return Ok(Value::Null);
         }
+
+        // active_signature may be negative value.
+        // So if it is negative value, we convert it into zero.
+        let active_signature_index = match help.active_signature {
+            Some(sig) => {
+                if sig.is_negative() {
+                    0
+                } else {
+                    sig as u64
+                }
+            }
+            None => 0,
+        }
+        .to_usize()?;
+
         let active_signature = help
             .signatures
-            .get(help.active_signature.unwrap_or(0).to_usize()?)
+            .get(active_signature_index)
             .ok_or_else(|| err_msg("Failed to get active signature"))?;
+
+        // active_signature may be negative value.
+        // So if it is negative value, we convert it into zero.
+        let active_parameter_index = match help.active_parameter {
+            Some(param) => {
+                if param.is_negative() {
+                    0
+                } else {
+                    param as u64
+                }
+            }
+            None => 0,
+        }
+        .to_usize()?;
+
         let active_parameter: Option<&ParameterInformation>;
         if let Some(ref parameters) = active_signature.parameters {
-            active_parameter = parameters.get(help.active_parameter.unwrap_or(0).to_usize()?);
+            active_parameter = parameters.get(active_parameter_index);
         } else {
             active_parameter = None;
         }
```

Comment:
Or more compactly one could write this all as:

```rust
        let active_signature_index =
            (help.active_signature.unwrap_or(0).max(0) as u64).to_usize()?;
        let active_signature = help
            .signatures
            .get(active_signature_index)
            .ok_or_else(|| err_msg("Failed to get active signature"))?;

        let active_parameter_index =
            (help.active_parameter.unwrap_or(0).max(0) as u64).to_usize()?;
        let active_parameter = active_signature
            .parameters
            .as_ref()
            .and_then(|parameters| parameters.get(active_parameter_index));
```

(which I mostly did as an exercise for myself rather than as any criticism of your implementation)